### ISSUE TITLE
FEATURE: Quick Inventory Snapshot

### ISSUE THREAD

**Issue by marker5a** [Brewtarget/brewtarget#761]
Just thinking out loud about a nice to have.  I feel there might be a more convenient way to show what is in inventory w/ hops, fermentables, etc... all in one view.  

For isntance, you could go to the View menu bar at the top and there'd be a new entry called "In stock Invent." and that window would just show what's in stock.  It could still allow you to edit inventory, but would only show what's currently in stock.

Thoughts?

**Comment by marker5a** [comment]
Also, FWIW, if there is interest and/or support to merge it, I'd love to work on it

**Comment by matty0ung** [comment]
Yes, improving how we handle inventory has been on my wish list for a long while.  If you want to extend the UI in the way you suggest, then, yes, go for it.

The only thing I'd say is not to count on making a lot of changes to the database or to the `NamedEntity` classes (ie the ones in the `src/model` directory as those are not far off having a lot of incoming change from the BeerJSON work.

**Comment by marker5a** [comment]
Oh yeah, I like that!  That's sort of what started this... going through
old grain and hops and trying to figure out what to brew next :)

**Comment by matty0ung** [comment]
It should be easier to do once we've done the BeerJSON changes.

At the moment, for historical reasons, each Recipe holds a copy of an ingredient to signify the addition of that ingredient to the recipe.  I'm probably going to fix this as part of the BeerJSON work (literally what I'm currently working on) so that we have a proper "use of" record.  That probably involves extending `fermentable_in_recipe` and similar tables to hold more info so that we can do away with the duplication of records in the `fermentable` table (and probably drop the `fermentable_children` table altogether).

Once that's done, there would then be a strong case (I think) for doing something similar with inventory, so that we can track purchase dates, use-by dates and so forth for stock.  It would likely be similar work, but extending the `fermentable_in_inventory` table.  Hopefully we'd be able to get a leg-up from the BeerJSON work, if you see what I mean.

**Comment by marker5a** [comment]
I'm wondering, just because I'm looking at my hop inventory in the freezer, if it makes sense to add a toggle-able checkbox on each of the view pages that allows you to check it to only show inventory?  This would be either in lieu of, or in parallel to, having a seperate inventory page that has yeast, hops, fermentables and misc all in one window.

![image](https://github.com/Brewtarget/brewtarget/assets/6609377/462d1100-3747-492d-9169-f3771b625d53)

If that seems cool with you guys, I'll add that in and do a PR

**Comment by marker5a** [comment]
Perfect, I'll get cracking!

**Comment by matty0ung** [comment]
FYI, I am finally working on some significant changes to how inventory is stored and shown.  There is still a fair bit of work to do, but it's starting to come together.  At the moment I'm hoping it will not have a large impact on the "Fermentable Database" and similar windows.

**Comment by matty0ung** [comment]
> I'm wondering, just because I'm looking at my hop inventory in the freezer, if it makes sense to add a toggle-able checkbox on each of the view pages that allows you to check it to only show inventory? This would be either in lieu of, or in parallel to, having a seperate inventory page that has yeast, hops, fermentables and misc all in one window.
> 
> ![image](https://github.com/Brewtarget/brewtarget/assets/6609377/462d1100-3747-492d-9169-f3771b625d53)
> 
> If that seems cool with you guys, I'll add that in and do a PR

I have implemented this feature in my latest PR (https://github.com/Brewtarget/brewtarget/pull/1045).  It should be included in the next release.

**Comment by matty0ung** [comment]
Feature included in [5.0.0 release](https://github.com/Brewtarget/brewtarget/releases/tag/v5.0.0).
